gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaHealthModel.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.source.extractor.extract.kafka;
+
+import com.google.common.collect.EvictingQueue;
+import com.typesafe.config.Config;
+import java.util.concurrent.TimeUnit;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.gobblin.annotation.Alias;
+import org.apache.gobblin.source.extractor.extract.kafka.workunit.packer.KafkaTopicGroupingWorkUnitPacker;
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.util.event.ContainerHealthCheckFailureEvent;
+
+
+/**
+ * Class for modeling the current kafka ingestion health. This class is responsible for the maintaining the data and api's
+ * for checking on kafka ingestion health. This class is different from the {@link KafkaExtractorStatsTracker} because it
+ * contains the logic / definitions for whether a metric is considered "healthy"
+ */
+@Slf4j
+@Alias(value = "KafkaHealthModel")
+@Getter
+public class KafkaHealthModel {
+  public static final String KAFKA_INGESTION_HEALTH_PREFIX = "gobblin.kafka.healthCheck.";
```

Comment:
Not going to change this from healthcheck to health model since I don't want to break any existing customer configs (for little reason). 

I didn't see any overrides on our system personally, but it's better safe than sorry